gophish.go

```diff
@@ -79,3 +92,70 @@ func main() {
 	}()
 	wg.Wait()
 }
+
+func CheckAndCreateSSL(ListenURL string, CertPath string, KeyPath string, Logger *log.Logger) error {
+	// Check whether there is an existing SSL certificate and/or key, and if so, abort execution of this function
+	if _, err := os.Stat(CertPath); !os.IsNotExist(err) {
+		return nil
+	}
+	if _, err := os.Stat(KeyPath); !os.IsNotExist(err) {
+		return nil
+	}
+
+	Logger.Printf("Creating new self-signed certificates for administration interface...\n")
+
+	// RSA would be a perfectly fine alternative
+	priv, err := ecdsa.GenerateKey(elliptic.P384(), rand.Reader)
```

Comment:
I'm admittedly not a crypto expert by any means, but any particular reason we chose `P384` as opposed to `p-521`?